lib/tapioca/dsl/compilers/active_record_fixtures.rb

```diff
@@ -104,12 +106,53 @@ def method_names_from_eager_fixture_loader
 
         sig { params(mod: RBI::Scope, name: String).void }
         def create_fixture_method(mod, name)
-          mod.create_method(
+          model_class = active_record_models[name] || "T.untyped"
+          mod.create_method_with_sigs(
             name,
-            parameters: [create_rest_param("fixture_names", type: "T.any(String, Symbol)")],
-            return_type: "T.untyped",
+            sigs: [
+              mod.create_sig(
+                parameters: {
+                  fixture_name: "T.any(String, Symbol)",
+                  other_fixtures: "NilClass",
+                },
+                return_type: model_class,
+              ),
+              mod.create_sig(
+                parameters: {
+                  fixture_name: "T.any(String, Symbol)",
+                  other_fixtures: "T.any(String, Symbol)",
+                },
+                return_type: "T::Array[#{model_class}]",
+              ),
+            ],
+            parameters: [
+              RBI::ReqParam.new("fixture_name"),
+              RBI::RestParam.new("other_fixtures"),
+            ],
           )
         end
+
+        sig { returns(T::Hash[String, String]) }
+        def active_record_models
+          @active_record_models ||= T.let(
+            ActiveRecord::Base.descendants.map do |model|
+              [fixture_name(model.name), model.name]
+            end.to_h,
+            T.nilable(T::Hash[String, T::Class[ActiveRecord::Base]]),
+          )
+        end
+
+        sig { params(class_name: String).returns(String) }
+        def fixture_name(class_name)
```

Comment:
Since this compiler is for `ActiveRecord`,  is it safe to assume `ActiveSupport` is also loaded? It looks like we have a runtime dependency: https://rubygems.org/gems/activerecord/versions/7.1.3.2/dependencies.

Could we use https://api.rubyonrails.org/classes/String.html#method-i-singularize instead?

Reply:
Do we need to take consider custom inflections into account? If so then the `gsub` approach could fail.

Reply:
I'll look at how the rails code generates it and see if it uses the inflections.
And good call out on the ActiveSupport usage

Reply:
@andyw8 for the inflection, as far as I can tell, fixtures names are generated [here](https://github.com/rails/rails/blob/6d3fd5b98cc1c700fe9733521118abc7540546e7/activerecord/lib/active_record/fixtures.rb#L546) which seems to only depend on the configs (e.g. `ActiveRecord::Base`'s `pluralize_table_names` value.
Does that seem correct to you?

Reply:
@Morriar after some digging  I think we can use `ActiveRecord::FixtureSet.default_fixture_model_name` directly (that is what is used by the fixtures themselves).
Somehow sorbet complains about FixtureSet not being defined, it executes just fine though - any ideas on why that might be?